lib/pub_sub/breaker.rb

```diff
@@ -10,18 +10,18 @@ class Breaker
     class << self
 
       def execute(&block)
-        get_breaker.run do
-          begin
+        begin
+          get_breaker.run do
             block.call
-          rescue Exception => e
-            # Intercept for breaker's tracking purposes
-            PubSub.report_error e
-            raise # will be caught by the breaker once enough of these accumulate
           end
+        rescue CB2::BreakerOpen => e
+          PubSub.report_error e
+          on_breaker_open
+        rescue Exception => e
```

Comment:
The breaker run block already has built-in exception handling, this `rescue Exception => e` is to log each exception for the code block being executed within the breaker run block before it's rescued by the actual breaker.

Moving it down here would only rescue if the breaker itself has an exception, which I'm not sure we care to protect against here.